### PR TITLE
Hw5

### PR DESCRIPTION
**Pull request by iRezograf**
HW5 is done. Vladimir, please review it.

### CODE REVIEW

File: src/test/java/ru/training/at/hw5/RunAllCucumberAcceptanceTests.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package ru.training.at.hw5;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(features = "src/test/resources/hw5")
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw5/RunAllCucumberAcceptanceTests.java]: Please use runtime reference, something like `classpath:hw5`

File: src/test/java/ru/training/at/hw5/hook/WebDriverHook.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import ru.training.at.hw5.dp.TestContext;
+
+import static io.github.bonigarcia.wdm.config.DriverManagerType.CHROME;
+
+public class WebDriverHook {
+
+    public static String removeSpaces = "^ +| +$|( )+|(\\r\\n|\\n|\\r)";
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw5/hook/WebDriverHook.java]: This filed should be final and not in this class

File: src/test/java/ru/training/at/hw5/steps/UserTableStep.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        UserTablePageObj userTablePageObj;
+
+        userTablePageObj = new UserTablePageObj(
+                TestContext.getInstance()
+                        .getTestObject(TestContext.WEB_DRIVER));
+
+        List<List<String>> s = getActualUserTableList(userTablePageObj);
+        // to set identical structure with expected structure
+        // remove col with type (index[1])
+        for (int i = 0; i < 6; i++) {
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw5/steps/UserTableStep.java]: If the scenario contains not `6` rows, then there will be unexpected behavior

\+ same for the next loop

File: src/test/resources/conf.properties [hunk 4]
```diff
@@ -29,3 +33,30 @@ textFromLogList1 =  Colors: value changed to Yellow
...
+userTableName4 = Helen Bennett
+userTableName5 = Yoshi Tannamuri
+userTableName6 = Giovanni Rovelli
+
+userTableDescription1 = Wolverine
+userTableDescription2 = Spider Man
+userTableDescription3 = Vladzimir 
+userTableDescription4 = Captain America some description
+userTableDescription5 = Cyclope some description
+userTableDescription6 = Hulk some description
```
Comment (shoggy) [inline src/test/resources/conf.properties]: These are part of the scenario, and should not be here

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/ru/training/at/hw5/RunAllCucumberAcceptanceTests.java (hunk 1 above)
Comment (iRezograf) [reply src/test/java/ru/training/at/hw5/RunAllCucumberAcceptanceTests.java]: @CucumberOptions(features = "src/test/resources/hw5",
        tags = "@all",
        dryRun = false)

File: src/test/java/ru/training/at/hw5/steps/UserTableStep.java (hunk 3 above)
Comment (iRezograf) [reply src/test/java/ru/training/at/hw5/steps/UserTableStep.java]: // number of row and col should be equal to actual data
        int row = arg.size();
        int col = arg.get(0).size();

        // get Access to actual
        UserTablePageObj userTablePageObj;

        userTablePageObj = new UserTablePageObj(
                TestContext.getInstance()
                        .getTestObject(TestContext.WEB_DRIVER));

        List<List<String>> s = getActualUserTableList(userTablePageObj);
        // to set identical structure with expected structure
        // remove col with type (index[1])
        for (int i = 0; i < row; i++) {
            s.get(i).remove(1);
        }

        // In Scenario was error:
        // Hulksome description - without space
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                assertThat(s.get(i).get(j)).contains(arg.get(i).get(j));
            }
        }

File: src/test/resources/conf.properties (hunk 4 above)
Comment (iRezograf) [reply src/test/resources/conf.properties]: It was as alternative. I removed it.

**Comment by iRezograf** [comment]
Changes resolved

image

**Review by shoggy — APPROVED** [review]

**Pull request merged**